### PR TITLE
adding VASP2JSON.py script to tools 

### PR DESCRIPTION
**Pull request by ldwillia**
for conversion from VASP's OUTCAR/POSCAR to the .json file type used by FitSNAP.

### CODE REVIEW

File: tools/VASP2JSON.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+num_atoms_per_type = print_num_atoms_per_type(POSCAR_file)
+num_atom_types = len(num_atoms_per_type)
+order_atom_types = []
+listAtomTypes = []
+
+# Start parsing through OUTCAR looking for keywords that are assocaited with the
+# different values for the data needed, such as forces or positions
+
+with open(OUTCAR_file, "rt") as f2:
```
Comment (charlessievers) [inline tools/VASP2JSON.py:78]:
```
with open(OUTCAR_file, "rt") as f2:
    for i, line in enumerate(f2.readlines()):
        if "VRHFIN" in line:
            ....
```

File: tools/VASP2JSON.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                for n in range(0, totalAtoms):
+                    listAtomTypes.append(atomType)
+    # Look for number of atoms in configuration
+    if "number of ions" in line:
+        columns = line.split()
+        natoms = int(columns[11])
+    # Look for lattice vectors for configuration
+    if "direct lattice vectors" in line:
+        line = lines[i + 1]
+        columns = line.split()
```
Comment (charlessievers) [inline tools/VASP2JSON.py:102]:
```
x1, x2, x3 = [float(x) for x in line[i + 1].split()]
```
or we can set it just to a single var
```
lattice_x = [float(x) for x in line[i + 1].split()]
```
or we could get the entire lattice in one
```
lattice = [[float(x) for x in line[i + j].split()] for j in range(3)]
```

File: tools/VASP2JSON.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        lattice_x = [x1, x2, x3]
+        lattice_y = [y1, y2, y3]
+        lattice_z = [z1, z2, z3]
+
+        all_lattice = [lattice_x, lattice_y, lattice_z]
+    # Look for stresses for configuration.  Assumes total stress is 14 lines down
+    # from where FORCE on cell is found
+    if "FORCE on cell" in line:
+        line = lines[i + 14]
+        columns = line.split()
```
Comment (charlessievers) [inline tools/VASP2JSON.py:126]: If we want to get crazy.
```
indices = [[2, 5, 7], [5, 3, 6], [7, 6, 4]]
stress = [[float(x) for x in list(map(line[i + 14].split().__getitem__, indices[j]))] for j in range(3)]
```

**Review by charlessievers — COMMENTED** [review]
I've left a few suggestions just for food for thought.

File: tools/VASP2JSON.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        z1 = float(columns[0])
+        z2 = float(columns[1])
+        z3 = float(columns[2])
+
+        lattice_x = [x1, x2, x3]
+        lattice_y = [y1, y2, y3]
+        lattice_z = [z1, z2, z3]
+
+        all_lattice = [lattice_x, lattice_y, lattice_z]
```
Comment (charlessievers) [inline tools/VASP2JSON.py:121]:
```
x1, x2, x3 = [float(x) for x in line[i + 1].split()]
```
or we can set it just to a single var
```
lattice_x = [float(x) for x in line[i + 1].split()]
```
or we could get the entire lattice in one
```
lattice = [[float(x) for x in line[i + j].split()] for j in range(3)]
```

Comment (ldwillia) [reply tools/VASP2JSON.py:121]: Updating to the second method with 
lattice_x = [float(x) for x in lines[i+1].split()[0:3]]
...

File: tools/VASP2JSON.py [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+# Start parsing through OUTCAR looking for keywords that are assocaited with the
+# different values for the data needed, such as forces or positions
+
+with open(OUTCAR_file, "rt") as f2:
+    lines = f2.readlines()
+outcar_config_number = 1
+
+for i in range(0, len(lines)):
+    line = lines[i]
```
Comment (charlessievers) [inline tools/VASP2JSON.py:83]:
```
for i, line in enumerate(lines):
    ....
```
This is my suggestion for having these two lines in one.

File: tools/VASP2JSON.py [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+        atom_cords = []
+        atom_force = []
+        line = lines[i + 2]
+        count = 0
+        for count in range(1, natoms + 1):
+            columns = line.split()
+            x_cord = float(columns[0])
+            y_cord = float(columns[1])
+            z_cord = float(columns[2])
```
Comment (charlessievers) [inline tools/VASP2JSON.py:148]:
```
x, y, z = [float(x) for x in line[i + 1].split()]

File: tools/VASP2JSON.py [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+            f_x = float(columns[3])
+            f_y = float(columns[4])
+            f_z = float(columns[5])
+
+            coordinates = [x_cord, y_cord, z_cord]
+            forces = [f_x, f_y, f_z]
+
+            atom_cords.append(coordinates)
+            atom_force.append(forces)
```
Comment (charlessievers) [inline tools/VASP2JSON.py:157]:
```
atom_cords.append([float(x) for x in line.split()[:3]])
atom_force.append([float(f) for f in line.split()[3:6]])
```

**Review by charlessievers — APPROVED** [review]
The script is great. My comments are merely suggestions.

**Review by mitwood — APPROVED** [review]

**Pull request merged**